mitodl/mit-open
# 191 back end logout flow

# What are the relevant tickets?
https://github.com/mitodl/mit-open/issues/191

# Description (What does it do?)
This PR solves the issue described in this RFC: https://docs.google.com/document/d/17tJ-C2EwWoSpJWZKjuhMVgsqGtyPH0IN9KakXvSKU0M/edit#heading=h.96o2gm3cfv2 

This also enables the future implementation of the flow here: https://docs.google.com/document/d/17tJ-C2EwWoSpJWZKjuhMVgsqGtyPH0IN9KakXvSKU0M/edit#heading=h.qr1x27po61aq

# How can this be tested?
These instructions assume that you have a local instance of Keycloak and MIT Open running and connected via an Open ID client in Keycloak.

Configure the MIT Open client in Keycloak:

1. The client can match the existing `ol-open-discussions-local` client found in the olapps realm in QA.
2. This client must have `Backchannel logout URL` set to `http://open.ol.local:8063/backend-logout/`


1. Login to MIT Open via http://open.ol.local:8063/login/ol-oidc
3. You should now be logged into Keycloak (verify by visiting http://localhost:8080/admin/master/console/)
4. You should now be logged into MIT Open (verify by visiting http://open.ol.local:8063/admin)
5. In a separate browser or incognito window, perform step 1, 2, and 3 again.
6. In one of the browser windows, navigate to http://open.ol.local:8063/logout.
7. You should now be logged out of Keycloak in only one of the browsers (verify by visiting http://localhost:8080/admin/master/console/)
8. Perform step 6 in the browser not originally used to complete step 6.
9. You should now be logged out of MIT Open both browsers (verify by visiting http://open.ol.local:8063/admin)
